### PR TITLE
Form parsing honors maxMemoryBuffer (bp #10543 #10565)

### PR DESCRIPTION
**Pull request by ignasi35**

### CODE REVIEW

File: documentation/manual/releases/release27/migration27/Migration27.md [hunk 1]
```diff
@@ -100,6 +100,28 @@ Other methods that were added to improve Java API:
...
 The API for body parser was mixing `Integer` and `Long` to define buffer lengths which could lead to overflow of values. The configuration is now uniformed to use `Long`. It means that if you are depending on `play.api.mvc.PlayBodyParsers.DefaultMaxTextLength` for example, you then need to use a `Long`. As such, `play.api.http.ParserConfiguration.maxMemoryBuffer` is now a `Long` too.
 
+### Parser `maxMemoryBuffer` limits
+
+Some payloads expand in memory when being parsed. So, the memory representation takes more space than the plaintext representation read from the wire. JSON is one of these formats. In order to prevent attacks that could lead to out of memory errors causing Denial-of-Service, body parsing and form binding must honour the `play.http.parser.maxMemoryBuffer` setting.
+
+It is also possible to relax the `maxMemoryBuffer` in specific cases. It is possible the JSON representation and the expanded representation differ in size and you need to use different limits. You can use a form binding with a customized limit using:
+
+```scala
+class MyController @Inject()(cc: MessagesControllerComponents) {
```
Comment (wsargent) [inline documentation/manual/releases/release27/migration27/Migration27.md:110]:
```suggestion
class MyController @Inject()(cc: MessagesControllerComponents) extends MessagesAbstractController(cc) {
```

File: documentation/manual/releases/release27/migration27/Migration27.md [hunk 2]
```diff
@@ -100,6 +100,28 @@ Other methods that were added to improve Java API:
...
+  ...
+  // create a new formBinding instance with increased limit 
+  val formBinding = cc.parsers.formBinding(300*1024) // limit to 300KiB
```
Comment (wsargent) [inline documentation/manual/releases/release27/migration27/Migration27.md:113]:
```suggestion
override implicit val formBinding = cc.parsers.formBinding(300*1024) // limit to 300KiB
```

File: documentation/manual/releases/release27/migration27/Migration27.md [hunk 3]
```diff
@@ -100,6 +100,28 @@ Other methods that were added to improve Java API:
...
+  val formValidationResult = form.bindFromRequest()(request, formBinding)
+  ...
+}
+```
+
+Controllers will always have a `FormBinding` instance build to honor the `play.http.parser.maxMemoryBuffer`. If you use the Forms from code outside a Controller, you may need to provide a `FormBinding`. For example, is you write unit tests you can use a `FormBinding` provided in `play.api.data.FormBinding.Implicits._` which uses a hardcoded limit which is good enough for tests. Add the implicit in scope:
```
Comment (wsargent) [inline documentation/manual/releases/release27/migration27/Migration27.md:119]:
```suggestion
Controllers will always have a `FormBinding` instance build to honor the `play.http.parser.maxMemoryBuffer`. If you use the `Forms` from code outside a Controller, you may need to provide a `FormBinding`. For example, if you write unit tests you can use a `FormBinding` provided in `play.api.data.FormBinding.Implicits._` which uses a hardcoded limit which is good enough for tests. Add the implicit in scope: 
```

File: documentation/manual/releases/release27/migration27/Migration27.md (hunk 1 above)
Comment (ignasi35) [reply documentation/manual/releases/release27/migration27/Migration27.md:110]: That was a wrong copy/paste. It should demo `MessagesControllerComponents`.

File: documentation/manual/releases/release27/migration27/Migration27.md (hunk 2 above)
Comment (ignasi35) [reply documentation/manual/releases/release27/migration27/Migration27.md:113]: The writeup was ambiguous. This was meant to demo the creation of a field inside a method to overwrite a single bind invocation, not a controller-wide override.

I've rewriten the example.

**Review by octonato — APPROVED** [review]
LGTM

**Pull request merged**